src/scep/Client/envelope.py

```diff
@@ -117,10 +133,16 @@ def _build_recipient_info(self, symmetric_key, recipient):
         Returns:
               RecipientInfo: Instance of ASN.1 data structure with required attributes and encrypted key.
         """
-        encrypted_symkey = recipient.public_key.encrypt(
-            plaintext=symmetric_key,
-            padding_type='pkcs'
-        )
+        if self._KeyEncryptionAlgorithm.native['algorithm'] == 'rsaes_oaep' :
+            encrypted_symkey = recipient.public_key.encrypt(
+                plaintext=symmetric_key,
+                padding_type='oaep'
+            )
+        else :
```

Comment:
can we add a check for `rsa` type and throw error if any other type of algorithm is used.